zerver/lib/actions.py

```diff
@@ -2229,120 +2229,23 @@ def do_create_realm(string_id, name, restricted_to_domain=None,
-    if log:
-        log_event(event)
-    send_event(event, [user_profile.id])
-
-
-def do_change_enable_sounds(user_profile, enable_sounds, log=True):
-    # type: (UserProfile, bool, bool) -> None
-    user_profile.enable_sounds = enable_sounds
-    user_profile.save(update_fields=["enable_sounds"])
-    event = {'type': 'update_global_notifications',
-             'user': user_profile.email,
-             'notification_name': 'enable_sounds',
-             'setting': enable_sounds}
-    if log:
-        log_event(event)
-    send_event(event, [user_profile.id])
-
-def do_change_enable_offline_email_notifications(user_profile, offline_email_notifications, log=True):
-    # type: (UserProfile, bool, bool) -> None
-    user_profile.enable_offline_email_notifications = offline_email_notifications
-    user_profile.save(update_fields=["enable_offline_email_notifications"])
-    event = {'type': 'update_global_notifications',
-             'user': user_profile.email,
-             'notification_name': 'enable_offline_email_notifications',
-             'setting': offline_email_notifications}
-    if log:
-        log_event(event)
-    send_event(event, [user_profile.id])
-
-def do_change_enable_offline_push_notifications(user_profile, offline_push_notifications, log=True):
-    # type: (UserProfile, bool, bool) -> None
-    user_profile.enable_offline_push_notifications = offline_push_notifications
-    user_profile.save(update_fields=["enable_offline_push_notifications"])
-    event = {'type': 'update_global_notifications',
-             'user': user_profile.email,
-             'notification_name': 'enable_offline_push_notifications',
-             'setting': offline_push_notifications}
-    if log:
-        log_event(event)
-    send_event(event, [user_profile.id])
-
-def do_change_enable_online_push_notifications(user_profile, enable_online_push_notifications, log=True):
-    # type: (UserProfile, bool, bool) -> None
-    user_profile.enable_online_push_notifications = enable_online_push_notifications
-    user_profile.save(update_fields=["enable_online_push_notifications"])
-    event = {'type': 'update_global_notifications',
-             'user': user_profile.email,
-             'notification_name': 'enable_online_push_notifications',
-             'setting': enable_online_push_notifications}
-    if log:
-        log_event(event)
-    send_event(event, [user_profile.id])
-
-def do_change_enable_digest_emails(user_profile, enable_digest_emails, log=True):
-    # type: (UserProfile, bool, bool) -> None
-    user_profile.enable_digest_emails = enable_digest_emails
-    user_profile.save(update_fields=["enable_digest_emails"])
+def do_change_notification_settings(user_profile, name, value, log=True):
+    # type: (UserProfile, str, bool, bool) -> None
+    """Takes in a UserProfile object, the name of a global notification
+    preference to update, and the value to update to
+    """
 
-    if not enable_digest_emails:
```

Comment:
I'd like to call out this chunk of code I deleted here.

In the previous model, the `do_change_enable_digest_emails` method was slightly different from the rest of the notification preference setters because it cleared the queue of emails backlogged for a user.

I moved this logic from the model layer to the view layer (`zerver/views/unsubscribe:39`) to keep the model layer only concerned with data setting, as well as allowing the digest preference method to fit modularly in with the rest of the notification preference setters.

If you think the queue clearing should stay coupled with the data change in the model layer I'm fine with that and can keep the digest preference setter its own function - I just wanted to call this out as a more significant "change" I made to get approval.